lib/rgb_matrix/engine.ex

```diff
@@ -96,64 +100,76 @@ defmodule RGBMatrix.Engine do
     %State{state | paintables: paintables}
   end
 
-  defp set_animation(state, animation) do
+  defp set_animation(state, animation_type) do
+    {render_in, animation} = Animation.new(animation_type, state.leds)
+
+    state = schedule_next_render(state, render_in)
+
     %State{state | animation: animation}
   end
```

Comment:
There are two `set_animation` functions in this module. One private (this one) and one public, defined on L48. I realize it's a different arity, but having two identically-named functions that do different things and have different visibility seems like a recipe for mistakes.